src/sdks/core/src/cpp/sdk/cpptest/unit/capabilitiesTest.cpp

```diff
@@ -0,0 +1,156 @@
+#include "unit.h"
+
+
+class CapabilitiesTest : public ::testing::Test {
+	protected:
+		JsonEngine* jsonEngine;
+        Firebolt::Error error = Firebolt::Error::None;
+
+	void SetUp() override
+	{
+		jsonEngine = new JsonEngine();
+	}
+
+	void TearDown() override
+	{
+		delete jsonEngine;
+	}
+};
+
+
+TEST_F(CapabilitiesTest, Supported)
+{
+	nlohmann::json_abi_v3_11_3::json expectedValues = nlohmann::json::parse(jsonEngine->get_value("Capabilities.supported"));
+
+
+	bool supported = Firebolt::IFireboltAccessor::Instance().CapabilitiesInterface().supported("example", &error);
+
+	EXPECT_EQ(error, Firebolt::Error::None);
+	EXPECT_EQ(supported, expectedValues);
+}
+
+
+TEST_F(CapabilitiesTest, CapabilitiesSupported)
+{
+    nlohmann::json_abi_v3_11_3::json expectedValues = nlohmann::json::parse(jsonEngine->get_value("Capabilities.supported"));
+    
+    std::string capability = "xrn:firebolt:capability:wifi:scan";
+    
+    bool status = Firebolt::IFireboltAccessor::Instance().CapabilitiesInterface().supported(capability, &error);
+
+    EXPECT_EQ(error, Firebolt::Error::None);
+    EXPECT_EQ(status, expectedValues);
+}
+
+TEST_F(CapabilitiesTest, Available)
+{
+    nlohmann::json_abi_v3_11_3::json expectedValues = nlohmann::json::parse(jsonEngine->get_value("Capabilities.available"));
+    
+    std::string capability = "xrn:firebolt:capability:token:device";
+   
+    bool status = Firebolt::IFireboltAccessor::Instance().CapabilitiesInterface().available(capability, &error);
+
+    EXPECT_EQ(error, Firebolt::Error::None);
+    EXPECT_EQ(status, expectedValues);
+}
+
+
+
+TEST_F(CapabilitiesTest, Permitted)
+{
+    nlohmann::json expectedValues = nlohmann::json::parse(jsonEngine->get_value("Capabilities.permitted"));
+    
+    std::string capability = "xrn:firebolt:capability:input:keyboard";
+
+    // Initialize options with a role
+    std::optional<Firebolt::Capabilities::CapabilityOption> options;
+    options.emplace();  // Initialize the optional
+    options->role = Firebolt::Capabilities::Role::MANAGE;
+
+    // Call the permitted function
+    bool status = Firebolt::IFireboltAccessor::Instance().CapabilitiesInterface().permitted(capability, options, &error);
+
+    EXPECT_EQ(error, Firebolt::Error::None);
+    EXPECT_EQ(status, expectedValues);
+}
+
+TEST_F(CapabilitiesTest, Granted)
+{
+    nlohmann::json expectedValues = nlohmann::json::parse(jsonEngine->get_value("Capabilities.granted"));
+    std::string capability = "xrn:firebolt:capability:localization:postal-code";
+
+    std::optional<Firebolt::Capabilities::CapabilityOption> options;
+    bool status = Firebolt::IFireboltAccessor::Instance().CapabilitiesInterface().granted(capability, options, &error);
+   
+    EXPECT_EQ(error, Firebolt::Error::None);
+    EXPECT_EQ(status, expectedValues);
+}
+
+
+TEST_F(CapabilitiesTest, TestCapabilitiesinfo) {
```

Comment:
Can we have some more clarity through comments on what's being tested here?

Reply:
this still needs to be addressed. but somehow became outdate. will keep this open until being addressed